base/indices.jl

```diff
@@ -231,6 +231,9 @@ function setindex_shape_check(X::AbstractArray{<:Any,2}, i::Integer, j::Integer)
     end
 end
 
+setindex_shape_check(::Any...) =
+    throw(ArgumentError("Scalar assignment to a vector is not supported: maybe you wanted to use `.=`?"))
```

Comment:
What about this alternative?  Would this make sense?  Or is it simply too pedantic?

```suggestion
    throw(ArgumentError("indexed assignment of a single value to many locations is not supported; perhaps use broadcasting `.=` instead?"))
```